bin/worker.js

```diff
@@ -37,32 +40,63 @@ export default {
     // Timestamp to measure file retrieval performance (from cache and from SP)
     const fetchStartedAt = performance.now()
 
-    const { response, cacheMiss, contentLength } = await retrieveFile(
+    const { response, cacheMiss } = await retrieveFile(
       BASE_URL,
       pieceCid,
       env.CACHE_TTL,
     )
 
+    if (!response.body) {
+      return new Response(
+        JSON.stringify({
+          error: 'Upstream response has no readable body.',
+          reason:
+            'The storage provider returned a response without a body, which cannot be streamed or measured for egress.',
+          upstreamUrl: request.url,
+          status: 502,
+        }),
+        {
+          status: 502,
+          headers: { 'Content-Type': 'application/json' },
+        },
+      )
+    }
+
+    // Stream and count bytes
+    // We create two identical streams, one for the egress measurement and the other for returning the response as soon as possible
+    const streams = response.body.tee()
```

Comment:
I am a bit concerned about duplicating the response stream
1. What is the performance & memory overhead? 
2. In the future, we will want to measure time to last byte (from worker execution start to the time we sent back the last byte), I think that won't be possible using the `tee` approach.

Having wrote that, I think the current version is good enough for now, and we can iterate on better approaches later.

WDYT, @juliangruber?